public/app/features/live/pipeline/models.gen.ts

```diff
@@ -0,0 +1,116 @@
+  fields: Field[];
+}
+export interface Label {
+  name: string;
+  value: string;
+}
+export interface Field {
+  name: string;
+  type: number;
+  value: string;
+  labels?: Label[];
+  config?: FieldConfig;
+}
+export interface AutoJsonConverterConfig {
+  fieldTips?: { [key: string]: Field };
+}
+export interface ConverterConfig {
+  type: Omit<keyof ConverterConfig, 'type'>;
+  jsonAuto?: AutoJsonConverterConfig;
+  jsonExact?: ExactJsonConverterConfig;
+  influxAuto?: AutoInfluxConverterConfig;
+  jsonFrame?: JsonFrameConverterConfig;
+}
+export interface MultipleFrameProcessorConfig {
+  processors: FrameProcessorConfig[];
+}
+export interface KeepFieldsFrameProcessorConfig {
+  fieldNames: string[];
+}
+export interface DropFieldsFrameProcessorConfig {
+  fieldNames: string[];
+}
+export interface FrameProcessorConfig {
+  type: Omit<keyof FrameProcessorConfig, 'type'>;
+  dropFields?: DropFieldsFrameProcessorConfig;
+  keepFields?: KeepFieldsFrameProcessorConfig;
+  multiple?: MultipleFrameProcessorConfig;
+}
+
+export interface ChangeLogOutputConfig {
+  fieldName: string;
+  channel: string;
+}
+export interface LokiOutputConfig {
+  uid: string;
+}
+export interface RemoteWriteOutputConfig {
+  uid: string;
+  sampleMilliseconds: number;
+}
+export interface ThresholdOutputConfig {
+  fieldName: string;
+  channel: string;
+}
+export interface NumberCompareFrameConditionConfig {
+  fieldName: string;
+  op: string;
+  value: number;
+}
+export interface MultipleFrameConditionCheckerConfig {
+  conditionType: string;
+  conditions: FrameConditionCheckerConfig[];
+}
+export interface FrameConditionCheckerConfig {
+  type: Omit<keyof FrameConditionCheckerConfig, 'type'>;
+  multiple?: MultipleFrameConditionCheckerConfig;
+  numberCompare?: NumberCompareFrameConditionConfig;
+}
+export interface ConditionalOutputConfig {
+  condition?: FrameConditionCheckerConfig;
+  output?: FrameOutputterConfig;
+}
+export interface RedirectOutputConfig {
+  channel: string;
+}
+export interface ManagedStreamOutputConfig {}
+export interface FrameOutputterConfig {
+  type: Omit<keyof FrameOutputterConfig, 'type'>;
+  managedStream?: ManagedStreamOutputConfig;
+  multiple?: MultipleOutputterConfig;
+  redirect?: RedirectOutputConfig;
+  conditional?: ConditionalOutputConfig;
+  threshold?: ThresholdOutputConfig;
+  remoteWrite?: RemoteWriteOutputConfig;
+  loki?: LokiOutputConfig;
+  changeLog?: ChangeLogOutputConfig;
+}
+export interface MultipleOutputterConfig {
+  outputs: FrameOutputterConfig[];
+}
+
+export interface SubscriberConfig {
+  type: Omit<keyof SubscriberConfig, 'type'>;
+  multiple?: MultipleSubscriberConfig;
+}
+export interface MultipleSubscriberConfig {
+  subscribers: SubscriberConfig[];
+}
+
+export interface RedirectDataOutputConfig {
+  channel: string;
+}
+export interface DataOutputterConfig {
+  type: Omit<keyof DataOutputterConfig, 'type'>;
+  redirect?: RedirectDataOutputConfig;
+  loki?: LokiOutputConfig;
+}
```

Comment:
this file does not include the entire rule config -- a type that defines permissions, channel, etc

Reply:
done in 39289b676254595907b743c04c20f2a85887d7c6